data/assets/examples/modelshader/model_fs.glsl

```diff
@@ -54,22 +52,44 @@ uniform int nLightSources;
 uniform vec3 lightDirectionsViewSpace[8];
 uniform float lightIntensities[8];
 
-uniform float opacity = 1.0;
+uniform bool performManualDepthTest = false;
+uniform sampler2D gBufferDepthTexture;
+
+uniform vec2 resolution;
 
 Fragment getFragment() {
   Fragment frag;
+  frag.depth = vs_screenSpaceDepth;
+  frag.gPosition = vs_positionCameraSpace;
+  frag.gNormal = vec4(vs_normalViewSpace, 0.0);
+  frag.disableLDR2HDR = true;
+  frag.color.a = 1.0;
 
+  if (performManualDepthTest) {
+    // gl_FragCoord.x goes from 0 to resolution.x and gl_FragCoord.y goes from 0 to
+    // resolution.y, need to normalize it
+    vec2 texCoord = gl_FragCoord.xy;
+    texCoord.x = texCoord.x / resolution.x;
+    texCoord.y = texCoord.y / resolution.y;
+
+    // Manual depth test
+    float gBufferDepth = denormalizeFloat(texture(gBufferDepthTexture, texCoord).x);
+    if (vs_screenSpaceDepth > gBufferDepth) {
+      frag.color = vec4(0.0);
+      frag.depth = gBufferDepth;
+      return frag;
+    }
+  }
+
+  // Frag color is the normal values
```

Comment:
Unclear to me what is meant by "normal values". Remove this comment? Or rephrase